src/redis-cli.c

```diff
@@ -1486,7 +1486,12 @@ static int cliSendCommand(int argc, char **argv, long repeat) {
                 config.in_multi = 0;
                 config.input_dbnum = config.dbnum = config.pre_multi_dbnum;
                 cliRefreshPrompt();
-            } 
+            } else if (!strcasecmp(command,"reset") && argc == 1 &&
+                                     config.last_cmd_type != REDIS_REPLY_ERROR) {
+                config.in_multi = 0;
+                config.dbnum = 0;
```

Comment:
You need also set `config.db_num`.
Besides, I think `config.resp3` should also be reset.

Reply:
Thanks,
There is no `db_num` field I guess you meant `config.input_dbnum`.

Reply:
Exactly. I made a typo.